### PR TITLE
Wall collisions

### PR DESCRIPTION
**Pull request by hotblack86**

### CODE REVIEW

**Review by matharotheelf — APPROVED** [review]
Very good work if I may say so myself.

File: src/hero.py [hunk 1]
```diff
@@ -14,8 +14,16 @@ def __init__(self, game, x_pos, y_pos):
...
-        self.x += dx
-        self.y += dy
+        if not self.collide_with_walls(dx, dy):
+            self.x += dx
+            self.y += dy
+    
+    def collide_with_walls(self, dx=0, dy=0):
+        for wall in self.game.walls_sprites:
+            if wall.x == self.x + dx and wall.y == self.y + dy:
+                print("Wall collision")
```
Comment (CKKH) [inline src/hero.py:24]: MVP! 🎆

File: src/main.py [hunk 2]
```diff
@@ -38,11 +38,12 @@ def new(self):
         self.all_sprites = pg.sprite.Group()
         self.all_sprites.add(Enemy(1, 1))
         self.hero  = Hero(self, 5, 5)
+        self.walls_sprites = pg.sprite.Group()
         self.all_sprites.add(self.hero)
         for row, tiles in enumerate(self.map_data):
```
Comment (CKKH) [inline src/main.py:43]: I wonder if we can tuck this logic away into another method?

**Review by michaelnguyen974 — APPROVED** [review]

File: src/wall.py [hunk 3]
```diff
@@ -7,10 +7,13 @@
...
+    def __init__(self, game, x_pos, y_pos):
         pg.sprite.Sprite.__init__(self)
+        self.game = game
         self.image = pg.Surface((TILESIZE, TILESIZE))
         self.image.fill(BROWN)
         self.rect = self.image.get_rect()
+        self.x = x_pos
+        self.y = y_pos
         self.rect.x = TILESIZE * x_pos
         self.rect.y = TILESIZE * y_pos
```
Comment (CKKH) [inline src/wall.py:19]: I think we can remove lines 18 and 19 now that we have lines 16 and 17

**Review by CKKH — APPROVED** [review]
Awesome! Super super close to MVP, enemies next!

File: src/wall.py (hunk 3 above)
Comment (michaelnguyen974) [reply src/wall.py:19]: We need the self.x and self.y for collisions. 
rect.x and rect.y are needed for determining the size of the tiles

**Pull request merged**